### PR TITLE
[TASK] Add contribution guide

### PR DESCRIPTION
**Pull request by DanielSiepmann**
Resolves: #721

### CODE REVIEW

**Comment by DanielSiepmann** [comment]
@sbuerk would be cool if you could review the change, as it is based on your feedback from December.

**Review by sbuerk — APPROVED** [review]
This is at least good first start. This can be enriched later - or linking to repo internal enhanced documentation about some topics for the contribution.

Only one thing, I'm not really sure about, is the filename.

As this is markdown, it may be use-full to use CONTRIBUTING.md. TYPO3 core has that naming too.

https://github.com/TYPO3/typo3/blob/main/CONTRIBUTING.md

Approving for now, but please reconsider the file naming with your team.

**Comment by DanielSiepmann** [comment]
> As this is markdown, it may be use-full to use CONTRIBUTING.md. TYPO3 core has that naming too.

Thanks will adjust, https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors is misleading as it doesn't have a file extension. But linked examples also end with `.md`.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+1. Please check existing open and closed issues first and feel free to comment and re-open them.
+2. Use a meaningful title
+3. Provide enough context, a concrete use case and steps to reproduce bugs
+
+## Contributing code and documentation
+
+1. Create a fork of the repository
+2. Create a new branch holding your changes
+3. Apply your changes
+4. Commit your changes, follow the commit conventions: https://docs.typo3.org/m/typo3/guide-contributionworkflow/main/en-us/Appendix/CommitMessage.html#commitmessage
```
Comment (lukaszuznanski) [inline CONTRIBUTING.md:16]: this guide is directly for gerrit, should we mention it? Not all of those conventions are relevant for github

Comment (DanielSiepmann) [reply CONTRIBUTING.md:16]: Fair point, will adjust. I set PR to draft until I've finished.

**Pull request closed without merging**